### PR TITLE
Type fixes

### PR DESCRIPTION
**Pull request by TaylorBeeston**
I just copied over the plugin template ABC and then better typed each hook. Doing that, I found a couple small bugs and fixed those as well!

### CODE REVIEW

File: autogpt/plugin_template.py
Comment (BillSchumacher) [inline autogpt/plugin_template.py:1]: You can import this class from pip.

**Review by BillSchumacher — CHANGES_REQUESTED** [review]

File: autogpt/plugin_template.py
Comment (TaylorBeeston) [reply autogpt/plugin_template.py:1]: I didn't realize that guy was up already! Do you mind updating what's on pypi with the type changes that got merged in today?

**Comment by TaylorBeeston** [comment]
@BillSchumacher one thing I forgot in this PR: the `post_planning` function has the wrong type right now. As it is used, it is getting parsed JSON and should be returning parsed JSON, but the ABC type is expecting a string and returning a string. I'm not sure whether strings or dicts should be preferred here, so I wanted to get your input before making that decision

**Review by BillSchumacher — APPROVED** [review]

**Comment by BillSchumacher** [comment]
> @BillSchumacher one thing I forgot in this PR: the `post_planning` function has the wrong type right now. As it is used, it is getting parsed JSON and should be returning parsed JSON, but the ABC type is expecting a string and returning a string. I'm not sure whether strings or dicts should be preferred here, so I wanted to get your input before making that decision

No problem, thanks for the PR!

**Pull request merged**